packages/rs-drive-abci/src/asset_lock/fetch_tx_out.rs

```diff
@@ -1,138 +0,0 @@
-                        let transaction_data =
-                            match core.get_transaction_extended_info(&transaction_hash) {
-                                Ok(transaction) => transaction,
-                                Err(_e) => {
-                                    return Ok(ValidationResult::new_with_error(
-                                        BasicError::IdentityAssetLockTransactionIsNotFoundError(
-                                            IdentityAssetLockTransactionIsNotFoundError::new(
-                                                transaction_hash.as_hash().into_inner(),
-                                            ),
-                                        )
-                                        .into(),
-                                    ))
-                                }
-                            };
-
-                        if !transaction_data.chainlock {
-                            // Very weird
-                            // We are getting back that the transaction is not locked, but the chain
-                            // lock proof says that it should be, most likely the client is lying.
-                            // it would seem that the chain lock is malformed or we are under attack
-                            // todo: log this event
-                            // todo: ban the ip sending this request
-                            return Ok(ValidationResult::new_with_error(
-                                BasicError::InvalidIdentityAssetLockProofChainLockValidationError(
-                                    InvalidIdentityAssetLockProofChainLockValidationError::new(
-                                        transaction_hash,
-                                        asset_lock_proof.core_chain_locked_height,
-                                    ),
-                                )
-                                .into(),
-                            ));
-                        }
-                    }
-                }
-
-                let transaction = transaction_data.transaction().map_err(|e| {
-                    Error::Execution(ExecutionError::DashCoreConsensusEncodeError(e))
-                })?;
-                if let Some(tx_out) = transaction.output.get(output_index) {
-                    Ok(ValidationResult::new_with_data(tx_out.clone()))
-                } else {
-                    // Also seems to be a malformed asset lock
-                    // todo: log this event
-                    // todo: ban the ip sending this request
-                    Ok(ValidationResult::new_with_error(
-                        BasicError::IdentityAssetLockTransactionOutputNotFoundError(
-                            IdentityAssetLockTransactionOutputNotFoundError::new(output_index),
-                        )
-                        .into(),
-                    ))
-                }
-            }
-        }
-    }
-}
```

Comment:
I wouldn't like to mix functions of different logical levels inside the state transition structure

Reply:
@QuantumExplorer It fetches a transaction and it uses asset lock as a yet another requirement to do it, so it has nothing to do in asset lock's namespace, especially if it doesn't consume it and it's not a transformation;

It might also be a construction static method of Tx, but construction usually happens on some input data and not on IO, so in that case core won't be passed inside -- that's the second approach I see